ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -939,6 +939,164 @@ class EclCmdRun : public EclCmdWithEclTarget
     bool optPre64;  //only for troubleshooting, do not document
 };
 
+class EclCmdResults : public EclCmdCommon
+{
+public:
+    EclCmdResults() : optExceptionSeverity("info")
+    {
+    }
+    virtual eclCmdOptionMatchIndicator parseCommandLineOptions(ArgvIterator &iter)
+    {
+        if (iter.done())
+            return EclCmdOptionNoMatch;
+
+        for (; !iter.done(); iter.next())
+        {
+            const char *arg = iter.query();
+            if (*arg!='-')
+            {
+                if (optWuid.length())
+                {
+                    fprintf(stderr, "\nunrecognized argument %s\n", arg);
+                    return EclCmdOptionCompletion;
+                }
+                if (!looksLikeAWuid(arg, 'W'))
+                {
+                    fprintf(stderr, "\nargument should be a workunit id: %s\n", arg);
+                    return EclCmdOptionCompletion;
+                }
+                optWuid.set(arg);
+                continue;
+            }
+
+            if (iter.matchFlag(optNoRoot, ECLOPT_NOROOT))
+                continue;
+            if (iter.matchOption(optExceptionSeverity, ECLOPT_EXCEPTION_LEVEL))
+                continue;
+            if (iter.matchFlag(optPre64, "--pre64")) //only for troubleshooting, do not document
+                continue;
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return EclCmdOptionMatch;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (optWuid.isEmpty())
+        {
+            fprintf(stderr, "No WUID provided.\n");
+            return 0;
+        }
+
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+
+    int gatherLegacyServerResults(IClientWsWorkunits* client, const char *wuid)
+    {
+        Owned<IClientWUInfoRequest> req = client->createWUInfoRequest();
+        req->setWuid(wuid);
+        req->setIncludeExceptions(true);
+        req->setIncludeGraphs(false);
+        req->setIncludeSourceFiles(false);
+        req->setIncludeResults(false); //ECL layout results, not xml
+        req->setIncludeResultsViewNames(false);
+        req->setIncludeVariables(false);
+        req->setIncludeTimers(false);
+        req->setIncludeDebugValues(false);
+        req->setIncludeApplicationValues(false);
+        req->setIncludeWorkflows(false);
+        req->setIncludeXmlSchemas(false);
+        req->setIncludeResourceURLs(false);
+        req->setSuppressResultSchemas(true);
+        Owned<IClientWUInfoResponse> resp = client->WUInfo(req);
+        int ret = outputMultiExceptionsEx(resp->getExceptions());
+
+        IConstECLWorkunit &wu = resp->getWorkunit();
+        IArrayOf<IConstECLException> &exceptions = wu.getExceptions();
+        unsigned count = wu.getResultCount();
+        if (count<=0 && exceptions.ordinality()<=0)
```

Comment:
Checking ret or exceptions.ordinality() are equivalent here.  But it may be clearer to use ret.  I'll change it.